cli/cmd/diagnose_util/profiling_util.go

```diff
@@ -133,3 +134,100 @@ func captureProfile(ctx context.Context, client *kube.Client, podName string, na
 
 	return nil
 }
+
+func FetchOdigosCollectorMetrics(ctx context.Context, client *kube.Client, metricsDir string) error {
+	odigosNamespace, err := resources.GetOdigosNamespace(client, ctx)
+	if err != nil {
+		return nil
+	}
+
+	var wg sync.WaitGroup
+
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		err = collectMetrics(ctx, client, odigosNamespace, metricsDir, consts.CollectorsRoleClusterGateway)
+		if err != nil {
+			fmt.Printf("Error Getting Metrics Data of: %v, because: %v\n", consts.CollectorsRoleClusterGateway, err)
+		}
+	}()
+
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		err = collectMetrics(ctx, client, odigosNamespace, metricsDir, consts.CollectorsRoleNodeCollector)
+		if err != nil {
+			fmt.Printf("Error Getting Metrics Data of: %v, because: %v\n", consts.CollectorsRoleNodeCollector, err)
+		}
+	}()
+
+	wg.Wait()
+
+	return nil
+}
+
+func collectMetrics(ctx context.Context, client *kube.Client, odigosNamespace string, metricsDir string, collectorRole consts.CollectorRole) error {
+	collectorPods, err := client.CoreV1().Pods(odigosNamespace).List(ctx, metav1.ListOptions{
+		LabelSelector: "odigos.io/collector-role=" + string(collectorRole),
+	})
+	if err != nil {
+		return err
+	}
+
+	var wg sync.WaitGroup
+
+	for _, collectorPod := range collectorPods.Items {
+		fmt.Printf("Fetching metrics for pod: %v", collectorPod.Name)
+		metricFilePath := filepath.Join(metricsDir, collectorPod.Name)
+		metricFile, err := os.OpenFile(metricFilePath, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0666)
+		if err != nil {
+			fmt.Printf("Error creating file: %v, because: %v", metricFilePath, err)
+			continue
+		}
+		defer metricFile.Close()
+
+		wg.Add(1)
+
+		go func() {
+			defer wg.Done()
+			err = captureMetrics(ctx, client, collectorPod.Name, odigosNamespace, metricFile, collectorRole)
+			if err != nil {
+				fmt.Printf("Error Getting Metrics Data of: %v, because: %v\n", metricFile, err)
+			}
+		}()
+	}
+
+	wg.Wait()
+	return nil
+}
+
+func captureMetrics(ctx context.Context, client *kube.Client, podName string, namespace string, metricFile *os.File, collectorRole consts.CollectorRole) error {
+	portNumber := ""
+	if collectorRole == consts.CollectorsRoleClusterGateway {
+		portNumber = "8888"
+	} else if collectorRole == consts.CollectorsRoleNodeCollector {
+		portNumber = "55682"
```

Comment:
you have this as
`k8sutils/pkg/consts >> OdigosNodeCollectorOwnTelemetryPortDefault`